### PR TITLE
A suggested interface for making custom queries through libmaxminddb

### PR DESCRIPTION
**Pull request by martinslota**
The idea is just to discuss how the interface could look going forward. This is I think the best suggestion that I managed to come up with up until now.

### CODE REVIEW

File: src/mmdb.mli [hunk 1]
```diff
@@ -34,22 +34,90 @@ module Lookup_ip_error : sig
...
 (** Open an MMDB file and return a reference to it *)
 val open_file : Path.t -> (t, Open_file_error.t) result
 
+module Ip = Types.Ip
+
+module Ip_data : sig
+  type t
+end
+
+val look_up_ip : t -> Ip.t -> (Ip_data.t, Lookup_ip_error.t) result
```
Comment (Leonidas-from-XIV) [inline src/mmdb.mli]: I think being consistent with `lookup` would be useful.

File: src/mmdb.mli [hunk 2]
```diff
@@ -34,22 +34,90 @@ module Lookup_ip_error : sig
...
+    val of_string_list : string list -> t
+
+    val to_string_list : t -> string list
+  end
+
+  module Answer : sig
+    type t
+  end
+
+  val look_up : t -> Ip.t -> Query.t -> Answer.t Lookup_result.t
```
Comment (Leonidas-from-XIV) [inline src/mmdb.mli]: `lookup` just to stay consistent.

File: src/mmdb.mli [hunk 3]
```diff
@@ -34,22 +34,90 @@ module Lookup_ip_error : sig
...
+    type t
+
+    val of_string_list : string list -> t
+
+    val to_string_list : t -> string list
+  end
+
+  module Answer : sig
+    type t
+  end
```
Comment (Leonidas-from-XIV) [inline src/mmdb.mli]: I think this can be simplified to just `type answer`.

File: src/mmdb.mli (hunk 1 above)
Comment (martinslota) [reply src/mmdb.mli]: I'll try to come up with a better name that doesn't have this kind of dichotomy between nouns and verbs.

**Comment by martinslota** [comment]
Implementation is in #24

**Pull request closed without merging**